google-cloud-core/src/main/java/com/google/cloud/Timestamp.java

```diff
@@ -43,12 +47,32 @@
       new Timestamp(253402300799L, (int) TimeUnit.SECONDS.toNanos(1) - 1);
 
   /** Regexp to split timestamps into date-hour-minute-second and fractional second components. */
-  private static final Pattern FORMAT_REGEXP = Pattern.compile("([^\\.]*)(\\.\\d{0,9})?Z");
+  private static final Pattern FORMAT_REGEXP = Pattern.compile("([^.]*)(\\.\\d{0,9})?Z");
+  private static final String DATE_GROUP = "(\\d{4}-\\d{1,2}-\\d{1,2})";
```

Comment:
Technically we don't have to adhere to that here. We can be loser or stricter than that as long as we make sure that while calling the bigquery rpc we pass strings of the correct format. But its fine.